operator/pkg/tpath/tpath_test.go

```diff
@@ -91,6 +91,93 @@ a:
     - v3
     - v3_regex
     name: n2
+`,
+		},
+		{
+			desc:      "ModifyNthListEntry",
+			path:      `a.b.[1].list.[:v2]`,
+			value:     `v-the-second`,
+			wantFound: true,
+			want: `
+a:
+  b:
+  - name: n1
+    value: v1
+  - list:
+    - v1
+    - v-the-second
+    - v3_regex
+    name: n2
+`,
+		},
+		{
+			desc:      "ModifyNthLeafListEntry",
+			path:      `a.b.[1].list.[2]`,
+			value:     `v-the-third`,
+			wantFound: true,
+			want: `
+a:
+  b:
+  - name: n1
+    value: v1
+  - list:
+    - v1
+    - v2
+    - v-the-third
+    name: n2
+`,
+		},
+		{
+			desc:      "ExtendNthLeafListEntry",
+			path:      `a.b.[1].list.[3]`,
```

Comment:
i suggest making the final index 4 to test this case. we could hit this in practice if multiple list entries are added in one CLI command and the flag representation is a map at some point in the processing.